### PR TITLE
app-interact lab

### PR DESCRIPTION
**Pull request by molecula2788**

### CODE REVIEW

File: content/chapters/app-interact/lab/README.md [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+So, if we run xeyes under strace, we expect to see some communication on the Unix socket that is created at the beginning.
+
+```Bash
+strace -e 'trace=!poll' -e trace='socket,connect,recvmsg' xeyes |& grep -v '\-1 EAGAIN
+```
+
+![](media/strace_xeyes.gif)
+
+
+# D-Bus
```
Comment (teodutu) [inline content/chapters/app-interact/lab/README.md:47]: This one is tricky to do, but I think it would greatly help the general narrative of the lab: try to come up with a "story" that you want to tell throughout the lab so that each new section "flows" logically from the previous one. The way they are now, these sections look rather unrelated to each other.

File: content/chapters/app-interact/lab/README.md [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+- The newly started Firefox process detected that another instance of Firefox is already running
+- The newly started Firefox process sent a message to the existing running process, requesting it to open an url in a new tab
+
+Since we're talking about message passing between 2 processes, there's a chance that maybe D-Bus was involved. Let's check: we'll use a tool called `dbus-monitor` that will print all messsages passed through D-Bus.
+
+```Bash
+student@os:~$ dbus-monitor
+```
+
+Then, in another terminal, we'll run again `firefox www.google.com`.
```
Comment (teodutu) [inline content/chapters/app-interact/lab/README.md:247]:
```suggestion
Then, in another terminal, we'll run `firefox www.google.com` again.
```
nit

**Review by teodutu — CHANGES_REQUESTED** [review]
Up to now, the written text looks great. The comments are mostly cosmetic or not requesting actual changes.

File: content/chapters/app-interact/lab/content/arena.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+## Arena
+
+### D-Bus
+
+Use the `dbus` python bindings to get the computer's battery level using a python script.
+You can start from the documentation [here](https://dbus.freedesktop.org/doc/dbus-python/tutorial.html#).
```
Comment (Alex-deVis) [inline content/chapters/app-interact/lab/content/arena.md:6]: ```You can start from the documentation```
It would be better to provide a starting point - maybe  a skeleton.

File: content/chapters/app-interact/lab/support/time-server/client.c [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <time.h>
+#include <unistd.h>
+#include <endian.h>
+
+#define MAX_SIZE 8
+
```
Comment (teodutu) [inline content/chapters/app-interact/lab/support/time-server/client.c:13]:
```suggestion
```

**Review by teodutu — CHANGES_REQUESTED** [review]
The content looks really well. Some cosmetic changes are still needed and when they're done, you can squash and merge this:

- Capitalise the words in section and subsection titles
- Use punctuation in comments and start each sentence with a capital letter

I left a few inline comments about the issues above, but they're not exhaustive.

**Review by teodutu — APPROVED** [review]
Now it looks fine. Before you merge it, add the new sections to [`config.yaml`](https://github.com/open-education-hub/operating-systems/blob/master/config.yaml) so it can be rendered with Docusaurus.
